extension/WebJobs.Extensions.RabbitMQ/Trigger/RabbitMQListener.cs

```diff
@@ -89,27 +92,35 @@ public Task StartAsync(CancellationToken cancellationToken)
                 var input = new TriggeredFunctionData() { TriggerValue = ea };
                 FunctionResult result = await this.executor.TryExecuteAsync(input, cancellationToken).ConfigureAwait(false);
 
-                if (result.Succeeded)
-                {
-                    this.rabbitMQModel.BasicAck(ea.DeliveryTag, false);
-                }
-                else
+                if (!result.Succeeded)
                 {
-                    if (ea.BasicProperties.Headers == null || !ea.BasicProperties.Headers.ContainsKey(Constants.RequeueCount))
-                    {
-                        this.CreateHeadersAndRepublish(ea);
-                    }
-                    else
+                    ea.BasicProperties.Headers ??= new Dictionary<string, object>();
+                    ea.BasicProperties.Headers.TryGetValue(RequeueCountHeaderName, out object headerValue);
+                    int requeueCount = Convert.ToInt32(headerValue, CultureInfo.InvariantCulture) + 1;
+
+                    if (requeueCount >= 5)
```

Comment:
Looks good overall, but had a question. Is the choice of 5 as the requeue count limit arbitrary here?

Reply:
Let me check if there is documentation on this, and if it is consistent or differs with each extension.

Reply:
There is some documentation on retries on <https://learn.microsoft.com/azure/azure-functions/functions-bindings-error-pages#retries>, but the retry counts are not documented. The Kafka extension Readme file mentions the default to be 2, which Cosmos DB does not seem to have any retry in place. For many extensions, the retry count is configurable through 'max retry count' property in host.json or via `FixedDelayRetry` or `ExponentialBackoffRetry` attribute on triggered function.

Note that retry via republish is a better option that the function executor itself invoking the function many times, as in the former case, the republished message is likely to end up on a different function app instance if multiple of them are running at a time.